src/EventGrid/EventGrid/Domain/RemoveAzureEventGridDomain.cs

```diff
@@ -0,0 +1,104 @@
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.EventGrid.Models;
+using Microsoft.Azure.Commands.EventGrid.Utilities;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.EventGrid
+{
+    [Cmdlet(
+        VerbsCommon.Remove,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "EventGridDomain",
+        DefaultParameterSetName = DomainNameParameterSet,
+        SupportsShouldProcess = true),
+    OutputType(typeof(bool))]
+
+    public class RemoveAzureEventGridDomain : AzureEventGridCmdletBase
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = EventGridConstants.ResourceGroupNameHelp,
+            ParameterSetName = DomainNameParameterSet)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasResourceGroup)]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = EventGridConstants.DomainNameHelp,
+            ParameterSetName = DomainNameParameterSet)]
+        [ResourceNameCompleter("Microsoft.EventGrid/domains", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        [Alias("DomainName")]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = EventGridConstants.DomainResourceIdHelp,
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = EventGridConstants.DomainInputObjectHelp,
+            ParameterSetName = DomainInputObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSDomain InputObject { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (this.ShouldProcess(this.Name, $"Remove domain {this.Name} in resource group {this.ResourceGroupName}"))
```

Comment:
@ahamad-MS I would move the logic for parsing `this.ResourceGroupName` and `this.Name` to before this if-statement since they won't have a value if the user provides a value for `-ResourceId` or `-InputObject` until after this line is called and displayed to the user

Reply:
Thanks for the suggestion but I am assuming this is along the previous comments where I am not able to use indexer/property as out parameter. Hence, keeping it as is.

Reply:
This should be fixed if you use the `ResourceIdentifier` type as mentioned in the initial comment -- I think making at least this change is important since we don't want to display a message to the user that says "Remove domain  in resource group  ".

Reply:
My bad. There are two fixes requested here:

1. Moving the parsing above the if statement ---> this is done now. Somehow, I missed it in previous iteration.

2. Using ResourceIdentified for parsing the domain topic nested resource which we discussed offline after I tried it and proved to require additional changes to remove the provider as well as additional parameters that we don't believe needed. This part is kept as is and we will continue using our custom util for domain topic